src/cis/fam_cis_thallium_client.cpp

```diff
@@ -563,7 +561,8 @@ void Fam_CIS_Thallium_Client::wait_for_copy(void *waitObj) {
         throw CIS_Exception(FAM_ERR_INVALID, "Copy waitObj is null");
     }
 
-    waitObjIn->wait();
+    Fam_CIS_Thallium_Response cisResponse = waitObjIn->wait();
```

Comment:
Just happened to notice now during the review that the waitObjIn object has been dynamically allocated using the new tl::async_response() and has not been freed/deleted subsequently, which might lead to memory leaks. So, is it fine to add code that deletes this object here for all the four wait_for operations in this PR itself?

Reply:
Good catch,

Reply:
Looks good!